### PR TITLE
#1900 Workspace members invite fixes

### PR DESCRIPTION
**Pull request by stephanevieira75**
# #1744 & #1900 Workspace member can invite or delete pending-emails only

https://user-images.githubusercontent.com/36481167/151386558-2cc0a515-c367-41a3-aabe-568a99f6e55e.mp4

# #1901 Add a notice to warn users that the company has reached the maximum number of users it can invite

https://user-images.githubusercontent.com/36481167/152163765-12bd069f-b85c-4983-97e2-1e2dbd920e48.mp4

### CODE REVIEW

**Comment by RomaricMourgues** [comment]
@BenoitTallandier could you confirm this new rules:
- Admins AND members can invite users to the workspace or get invitation link (before only admins)
- Admins AND members can remove pending invitations or re-generate invitation link (before only admins)
- Only Admins can remove members from the workspace
- Only Admins can set an other member as admin or not
- Guests cannot see or invite users to the workspace

**Comment by BenoitTallandier** [comment]
Yes, I confirm these rules :)

File: twake/backend/node/src/services/user/web/types.ts [hunk 1]
```diff
@@ -65,9 +65,38 @@ export interface UserObject {
...
+  COMPANY_MEMBERS_LIMIT = "company:members_limit", // 100
+}
+
+export enum CompanyFeaturesEnum {
+  CHAT_GUESTS = "chat:guests",
+  CHAT_MESSAGE_HISTORY = "chat:message_history",
+  CHAT_MULTIPLE_WORKSPACES = "chat:multiple_workspaces",
+  CHAT_EDIT_FILES = "chat:edit_files",
+  CHAT_UNLIMITED_STORAGE = "chat:unlimited_storage",
+  COMPANY_LIMIT_NOT_REACHED = "company:limit_not_reached",
```
Comment (RomaricMourgues) [inline twake/backend/node/src/services/user/web/types.ts:79]: Alors j'ai réfléchis cette nuit on va faire un gros chercher remplacer:
COMPANY_LIMIT_NOT_REACHED -> COMPANY_INVITE_MEMBER
company:limit_not_reached -> company:invite_member

File: twake/backend/node/src/services/workspaces/web/routes.ts [hunk 2]
```diff
@@ -110,11 +111,29 @@ const routes: FastifyPluginCallback<{
...
+      throw fastify.httpErrors.forbidden("Only company member can perform this action");
+    }
+  };
+
   const checkUserIsWorkspaceAdmin = async (
     request: FastifyRequest<{ Params: WorkspaceUsersRequest }>,
   ) => {
     if (!request.currentUser.id) {
-      throw fastify.httpErrors.forbidden("Only workspace moderator can perform this action");
+      // TODO error auth msg
```
Comment (RomaricMourgues) [inline twake/backend/node/src/services/workspaces/web/routes.ts:135]: On peut enlever les TODOs ?

File: twake/backend/node/src/services/workspaces/web/routes.ts [hunk 3]
```diff
@@ -110,11 +111,29 @@ const routes: FastifyPluginCallback<{
     return workspaceUser;
   };
 
+  const checkUserHasCompanyMemberLevel = async (
+    request: FastifyRequest<{ Params: WorkspaceUsersRequest }>,
+  ) => {
+    if (!request.currentUser.id) {
+      // TODO error auth msg
```
Comment (RomaricMourgues) [inline twake/backend/node/src/services/workspaces/web/routes.ts:118]: On peut enlever les TODOs ?

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]

File: twake/backend/node/src/services/user/web/types.ts (hunk 1 above)
Comment (stephanevieira75) [reply twake/backend/node/src/services/user/web/types.ts]: Good :+1

**Pull request merged**